### DISCUSSION TITLE
Does not detect model despite having model files in the "blob" directory, and cannot pull model from ollama.

### DISCUSSION THREAD

**Discussion by yuchenxie4645** [open-webui/open-webui#2071] [General]
# Bug Report

Does not detect model despite having model files in the "blob" directory. 

## Description

Cannot pull model locally on ubuntu but since I had a second version of openwebui working on the windows version I just copied the model files over to ubuntu at the "blob" directory (/usr/share/ollama/.ollama/models/blobs)

**Bug Summary:**

Ubuntu open web ui cannot pull model from ollama (option is physically not there). Tried copying files from windows version with functioning model pulling. It does not work because the web ui does not detect the model files.  

**Steps to Reproduce:**

I followed the standardized installation procedure provided by Ollama, including installing docker engine and ollama. Everything was functioning other than the models.
I had a windows variant which is 100% working and is able to pull models.
Then I copied the files over to ubuntu variant (exact models) and it does not work. 

**Expected Behavior:**
I was expecting it to have the "pull model from ollama" section in the models categories in settings.
I also expected the model files to work right after, tried restarting computer and restarting and recreating a new container.

**Actual Behavior:**
Does not have the "pull model from ollama" section.
Model is NOT detected.

## Environment

- **Open WebUI Version: 1.123**
- Ubuntu 24.04 LTS
- Chrome

**Confirmation:**

- [ Yes] I have read and followed all the instructions provided in the README.md.
- [Yes ] I am on the latest version of both Open WebUI and Ollama.
- [ No] I have included the browser console logs.
- [ No] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**

None

**Docker Container Logs:**

None

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

Used docker, on ubuntu 24.04 LTS.

![no_model_pulling_openwebui](https://github.com/open-webui/open-webui/assets/169230870/8b3c1ab1-18d3-471d-8413-93c06c750556)
![model_no_detect_openwebui](https://github.com/open-webui/open-webui/assets/169230870/ec531492-7954-402e-a2b2-8675a7b11800)

**Comment by ACK-J** [comment]
I'm having the same issue on Linux

**Reply by yuchenxie4645** [reply]
Found a fix yet?

**Reply by ACK-J** [reply]
Deleting the container and running the one with ollama bundled fixed it for me 
docker run -d -p 3000:8080 --gpus=all -v ollama:/root/.ollama -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:ollama

**Reply by yuchenxie4645** [reply]
interesting, I will try that :D

**Reply by yuchenxie4645** [reply]
Yep it is fixed, I appreciate it so much dude :D

**Comment by arsaboo** [comment]
First, please make sure that Ollama is working properly before trying to configure WebUI. Looks like you have issues with Ollama right now and these are unrelated to WebUI.

**Reply by yuchenxie4645** [reply]
Elaborate on "making sure that ollama is working properly". My understanding is, going to the link http://127.0.0.1:11434, it will say that Ollama is running.
![ollama_is_running](https://github.com/open-webui/open-webui/assets/169230870/9a3ecbe5-7011-4d91-bf70-7937326d8cf2)

**Comment by ACK-J** [comment]
No issues with ollama itself that I can see at the moment

```
root@pc:/$ ollama run gemma
>>> are you working?
I am an AI language model and am not employed in the traditional sense. I 
am designed to assist and provide information to users through language 
processing and natural language understanding.
```

**Reply by yuchenxie4645** [reply]
Open WebUI physically does not detect the model whatsoever:
![model_no_found_openwebui](https://github.com/open-webui/open-webui/assets/169230870/210c5452-36d0-4753-bc07-4025511f6bbd)

**Comment by yuchenxie4645** [comment]
I have just fixed the issue. What fixed for me is deleting the old container and purging Ollama and other related services following [this guide on how to uninstall](https://github.com/ollama/ollama/blob/main/docs/linux.md), then installing it with the bundled version.

**Comment by CtrlAiDel** [comment]
I will try this later. You're all heroes! xD
